### ISSUE TITLE
Pijul is broken in every version - source repo disappeared

### ISSUE THREAD

**Issue by colemickens** [NixOS/nixpkgs#12924]
Here's the log: https://gist.github.com/9f04ad2893fe1ace2096

Looks like maybe the git tag was deleted? I'll try to send a PR to fix shortly and reach out to the Pijul folks about what happened.

**Comment by colemickens** [comment]
```
mkdir temp; cd temp
darcs clone https://pijul.org
darcs show tags
```

They removed the 0.1 tag, it seems. Will send a PR to bump to 0.2.

**Comment by pijul** [comment]
Wait. I'm not sure what this means. We just released Pijul 0.2, which is a complete rewrite, with a totally unrelated darcs repository.

(btw while we're super happy to be in NixOS, none of the authors submitted Pijul, nor considered it ready at version 0.1).

**Comment by colemickens** [comment]
@pijul There is a definition for `pijul 0.1` in nixpkgs right now.

It referenced the `0.1` tag in the darcs repo, which now no longer exists. This means `pijul` is currently broken for anyone who didn't install it before you changed repos.

I'm working on a fix to move to `0.2` but it looks like maybe the output directory structure has changed, so I need to wait for my rebuild to finish before I can send a PR.

**Comment by pijul** [comment]
Ok. The darcs repository http://pijul.org shares no patch with the previous one. Not only has the directory structure changed, even the programming language has changed.

We're now using Rust, and we do have a .nix file at the root of our repository. The github mirror has it too: https://github.com/pijul/pijul/blob/master/default.nix

**Comment by colemickens** [comment]
@pijul Can you please check in your Cargo.lock file? I can't add this to nixpkgs without it.

**Comment by pijul** [comment]
Done.

**Comment by colemickens** [comment]
I'm a bit impeded now because [this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/fetch-cargo-deps#L24) is too strict and `cd`'s to `$src` to look for the `Cargo.lock`.

**Comment by colemickens** [comment]
The WIP is here: https://github.com/colemickens/nixpkgs/commit/03effaa946a2f1d9c5563f7cb7caf482b9b6ab1b

Still suspect problem is here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/fetch-cargo-deps#L24

**Comment by puffnfresh** [comment]
@colemickens can we parameterise fetch-cargo-deps with a `buildDir` or something?

**Comment by colemickens** [comment]
Hm, it looks like `fetch-cargo-deps` already takes `$src` as a parameter `src=$(realpath $1)`. Maybe my suspicion is wrong?

**Comment by puffnfresh** [comment]
@vcunat @colemickens @pijul I have it working and packaged in #13027.

**Comment by colemickens** [comment]
This was fixed in #13027. Thanks @puffnfresh.
